### PR TITLE
Support for FreeBSD get_app_path

### LINKED ISSUE #889: FreeBSD Support
This application is fantastic. It took a lot of talent, skill and dedication to make this.

I have been able to get this running on aseprite after a few minor tweaks (recompiling openssl port without OpenSSL v2, #888).

aseprite

I am able to get it running if I open it from ./build/bin. However, trying to access it any other way, such as /usr/local/bin/aseprite (after make install), gives the gui.xml error.

2015-12-05-094740_240x65_scrot

I see a similar error at https://code.google.com/p/aseprite/issues/detail?id=84.  Before I dig too deep into it, is ./aseprite only executing when cwd'd inside ./build/bin by design (for the purpose of development)?

### PR DESCRIPTION
**Pull request by tony**
Fixes #889

Add support with and without procfs.

See also: https://www.libsdl.org/tmp/SDL/src/filesystem/unix/SDL_sysfilesystem.c

### CODE REVIEW

File: src/base/fs_unix.h [hunk 1]
```diff
@@ -120,16 +122,23 @@ std::string get_app_path()
...
+  size_t size = path.size();
+  const int mib[] = { CTL_KERN, KERN_PROC, KERN_PROC_PATHNAME, -1 };
+  while (sysctl(mib, 4, &path[0], &size, NULL, 0) == -1)
+      path.resize(size);
+#endif
 
-#else
-
+if (!path[0] && (access("/proc", F_OK) == 0)) {
+#if defined(__FREEBSD__)
```
Comment (dacap) [inline src/base/fs_unix.h]: Why `__FREEBSD__` instead of `__FreeBSD__`? Is it `__FreeBSD__` or `__FREEBSD__`?

File: src/base/fs_unix.h [hunk 2]
```diff
@@ -120,16 +122,23 @@ std::string get_app_path()
...
+#elif __FreeBSD__
+  size_t size = path.size();
+  const int mib[] = { CTL_KERN, KERN_PROC, KERN_PROC_PATHNAME, -1 };
+  while (sysctl(mib, 4, &path[0], &size, NULL, 0) == -1)
+      path.resize(size);
+#endif
 
-#else
-
+if (!path[0] && (access("/proc", F_OK) == 0)) {
```
Comment (dacap) [inline src/base/fs_unix.h]: Is this required in OS X? why isn't this inside `__FreeBSD__`? Also this line needs a correct indentation.

**Comment by tony** [comment]
@dacap better now

**Comment by dacap** [comment]
Yeah, now it looks better :smile: Thanks for the patch!

**Pull request merged**